add global auth cookie

devadaviddd opened the pull request.

vhpx commented on apps/web/src/constants/common.ts:
```diff
@@ -2,6 +2,7 @@ import { version } from '../core/version';
 
 export const APP_VERSION = version;
 export const DEV_MODE = process.env.NODE_ENV === 'development';
+export const AUTH_DOMAIN = process.env.NEXT_PUBLIC_AUTH_DOMAIN || 'tuturuuu.com'
```
```suggestion
export const AUTH_COOKIE_DOMAIN = process.env.NEXT_PUBLIC_AUTH_COOKIE_DOMAIN
```

vhpx commented on apps/web/src/pages/_app.tsx:
```diff
@@ -8,6 +8,7 @@ import Providers from '../components/common/Providers';
 import Analytics from '../components/common/Analytics';
 import usePersistLocale from '../hooks/usePersistLocale';
 import { Database } from '../types/database.types';
+import { AUTH_DOMAIN, DEV_MODE } from '../constants/common';
```
```suggestion
import { AUTH_COOKIE_DOMAIN, DEV_MODE } from '../constants/common';
```

vhpx commented on apps/web/src/pages/_app.tsx:
```diff
@@ -17,7 +18,17 @@ export default function Application({
   // Create a new supabase browser client on every first render.
-  const [supabaseClient] = useState(() => createPagesBrowserClient<Database>());
+  const [supabaseClient] = useState(() =>
+    createPagesBrowserClient<Database>({
+      cookieOptions: {
+        domain: DEV_MODE ? undefined : AUTH_DOMAIN,
+        maxAge: 100000000,
+        path: '/',
+        sameSite: 'lax',
+        secure: true,
+      },
+    })
+  );
```
```suggestion
    createPagesBrowserClient<Database>(
    (DEV_MODE || !AUTH_COOKIE_DOMAIN)
    ? undefined
    : {
      cookieOptions: {
        domain: AUTH_COOKIE_DOMAIN,
        maxAge: 100000000,
        path: '/',
        sameSite: 'lax',
        secure: true,
      },
    })
  );
```

vhpx commented on turbo.json:
```diff
@@ -6,14 +6,13 @@
     "NEXT_PUBLIC_SUPABASE_URL",
     "SUPABASE_SERVICE_KEY",
-
+    "NEXT_PUBLIC_AUTH_DOMAIN",
```
```suggestion
    "NEXT_PUBLIC_AUTH_COOKIE_DOMAIN",
```

vhpx reviewed (changes requested):
Please take a look at the suggested changes. After it's resolved, I will merge this PR to main and staging so that we can test this feature out before rolling into production.

devadaviddd commented:
> Please take a look at the suggested changes. After it's resolved, I will merge this PR to main and staging so that we can test this feature out before rolling into production.

Thank you for the feedback on my pull request. I have reviewed your suggested changes and implemented them in a new commit. I believe I have addressed your comments - please take a look at the updated pull request when you have a chance. Let me know if you have any other changes you would like me to make. I appreciate you taking the time to provide suggestions to improve my code.
